x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/mapper/NativeRoleMappingStore.java

```diff
@@ -243,6 +243,7 @@ private void innerPutMapping(PutRoleMappingRequest request, ActionListener<Boole
                     .setId(getIdForName(mapping.getName()))
                     .setSource(xContentBuilder)
                     .setRefreshPolicy(request.getRefreshPolicy())
+                    .setWaitForActiveShards(request.getActiveShardCount())
```

Comment:
Do we need to pass the information through the `PutRoleMappingRequest`? It seems like we might as well use `NONE` here always. Seems good also for other uses of the store.

We always wait for the primary anyway during indexing.

Reply:
Sorry after I opened this PR, I actually thought of not including it on the request (to avoid unnecessary serialization changes) and simply passing it through as a separate param to `putRoleMapping`. 

> It seems like we might as well use NONE here always. Seems good also for other uses of the store. We always wait for the primary anyway during indexing. 

Just to validate: this is the case for stateful also? This API is exposed to stateful customers.

Reply:
I've pushed the suggested change; provided we wait for the primary on indexing in stateful, this change makes perfect sense to me.